Tests/ASTextNode2Tests.mm

```diff
@@ -91,20 +103,205 @@ - (void)testAccessibility
                 _textNode.defaultAccessibilityLabel, _attributedText.string);
 }
 
-- (void)testRespectingAccessibilitySetting
+- (void)testBasicAccessibilityWithExperiments
+{
+  [self setUpEnablingExperiments];
+
+  XCTAssertFalse(_textNode.isAccessibilityElement, @"Is not an accessiblity element as it's a UIAccessibilityContainer");
+  XCTAssertTrue(_textNode.accessibilityTraits == UIAccessibilityTraitStaticText,
+                @"Should have static text accessibility trait, instead has %llu",
+                _textNode.accessibilityTraits);
+  XCTAssertTrue(_textNode.defaultAccessibilityTraits == UIAccessibilityTraitStaticText,
+                @"Default accessibility traits should return static text accessibility trait, "
+                @"instead returns %llu",
+                _textNode.defaultAccessibilityTraits);
+
+  XCTAssertTrue([_textNode.accessibilityLabel isEqualToString:_attributedText.string],
+                @"Accessibility label is incorrectly set to \n%@\n when it should be \n%@\n",
+                _textNode.accessibilityLabel, _attributedText.string);
+  XCTAssertTrue([_textNode.defaultAccessibilityLabel isEqualToString:_attributedText.string],
+                @"Default accessibility label incorrectly returns \n%@\n when it should be \n%@\n",
+                _textNode.defaultAccessibilityLabel, _attributedText.string);
+
+  XCTAssertTrue(_textNode.accessibilityElements.count == 1, @"Accessibility elements should exist");
+  XCTAssertTrue([[_textNode.accessibilityElements[0] accessibilityLabel] isEqualToString:_attributedText.string],
+                @"First accessibility element incorrectly returns \n%@\n when it should be \n%@\n",
+                [_textNode.accessibilityElements[0] accessibilityLabel], _textNode.accessibilityLabel);
+  XCTAssertTrue([[_textNode.accessibilityElements[0] accessibilityLabel] isEqualToString:_attributedText.string],
+                @"First accessibility element incorrectly returns \n%@\n when it should be \n%@\n",
+                [_textNode.accessibilityElements[0] accessibilityLabel], _textNode.accessibilityLabel);
```

Comment:
Use XCTAssertEqual and XCTAssertEqualObjects so we get usable errors, throughout this PR.